### DISCUSSION TITLE
Dead units

### DISCUSSION THREAD

**Discussion by axx0** [axx0/Civ2-clone#45] [General]
The original (Civ2&Tot) does not store info on dead units/casualties, except for:
- casualty count of unit types (settlers, warriors, etc.), which is stored in tribes section in .SAV
- casualty timeline at the end of each turn (which is not stored in .SAV and is reset each time you load a game)

In .SAV dead units are marked with negative XY location values and are overwritten by new units once they're built.

If the need arises we could speed up LINQ queries by storing dead units info in a separate list, which more or less seems to be how the original works.

Anyway just putting some of my findings here. (Some info I've gathered from existing info on the web (Allard Hofelt's and catfish's guide on SAV layout), others I've researched myself.)

**Comment by reubene** [comment]
This is really useful info, I think it would be useful to remove the dead units from the units list as in a long game they could really start to slow down the LINQ queries. 

Would be useful for reports, stats or even a detailed replay to have some sort of history structure to record battles & who the attack and defender were and who won. I started something like this for cities founded but not much progress there yet

**Comment by axx0** [comment]
So from the casualty timeline the game records info of dead units:
- type of unit
- year of death
- location (and nearest city marked with ~)
- enemy unit civ

![Clipboard01](https://user-images.githubusercontent.com/21365802/165115496-c654b2c6-d22d-46fd-b005-c0bd35f1618f.png)

Like I mentioned this info is not stored in SAV which means that the casualty timeline is recorded only in a single gaming session. 
But like you said we can make our own logic.
